src/main/java/io/reactiverse/pgclient/impl/PgConnectionFactory.java

```diff
@@ -86,7 +90,88 @@ public void close() {
     client.close();
   }
 
-  public void connect(Handler<? super CommandResponse<Connection>> completionHandler) {
+  public void create(Handler<? super CommandResponse<Connection>> completionHandler) {
+    connect(ar -> {
+      if (ar.succeeded()) {
+        SocketConnection conn = ar.result();
+        conn.initializeCodec();
+        conn.sendStartupMessage(username, password, database, completionHandler);
+      } else {
+        completionHandler.handle(CommandResponse.failure(ar.cause()));
+      }
+    });
+  }
+
+  public void connect(Handler<AsyncResult<SocketConnection>> handler) {
+    switch (sslMode) {
+      case DISABLE:
+        initiateConnection(false, handler);
+        break;
+      case ALLOW:
+        initiateConnection(false, ar -> {
+          if (ar.succeeded()) {
+            handler.handle(Future.succeededFuture(ar.result()));
+          } else {
+            initiateConnection(true, handler);
+          }
+        });
+        break;
+      case PREFER:
+        initiateConnection(true, ar -> {
+          if (ar.succeeded()) {
+            handler.handle(Future.succeededFuture(ar.result()));
+          } else {
+            initiateConnection(false, handler);
+          }
+        });
+        break;
+      case VERIFY_FULL:
+        if (hostnameVerificationAlgorithm == null || hostnameVerificationAlgorithm.isEmpty()) {
```

Comment:
can't we verify this before connecting ?

Reply:
The purpose here is to force the check to avoid unneccesary connection if SSL options(via NetClientOptions) are not configured correctly and break the restrictions of sslmode.